core/src/test/scala/org/apache/spark/scheduler/EventLoggingListenerSuite.scala

```diff
@@ -95,6 +95,30 @@ class EventLoggingListenerSuite extends SparkFunSuite with LocalSparkContext wit
     }
   }
 
+  test("Event logging with password redaction") {
+    val secretPassword = "secret_password"
+    val conf = getLoggingConf(testDirPath, None).set("spark.executorEnv.HADOOP_CREDSTORE_PASSWORD",
+      secretPassword)
+    sc = new SparkContext("local-cluster[2,2,1024]", "test", conf)
```

Comment:
This is a pretty expensive way of testing this. Why not just call the redaction method and make sure it's doing the right thing?